src/events/http.js

```diff
@@ -215,6 +216,69 @@ function httpWrapper(wrappedFunction) {
                 this, buildParams(url, options, patchedCallback)
             );
 
+            /**
+             * Wraps 'on' method in a response to capture data event.
+             * @param {Function} wrappedResFunction The wrapped end function
+             * @returns {Function} The wrapped function
+             */
+            function responseOnWrapper(wrappedResFunction) { // eslint-disable-line no-inner-declarations,max-len
+                return function internalResponseOnWrapper(resEvent, resCallback) {
+                    if (resEvent !== 'data' || typeof resCallback !== 'function') {
+                        return wrappedResFunction.apply(this, [resEvent, resCallback]);
+                    }
+                    const resPatchedCallback = (chunk) => {
+                        if (chunk) {
+                            const totalSize = chunks.reduce(
+                                (total, item) => item.length + total,
+                                0
+                            );
+                            if (totalSize + chunk.length <= MAX_HTTP_VALUE_SIZE) {
+                                chunks.push(typeof chunk === 'string' ? Buffer.from(chunk) : chunk);
+                            }
+                        }
+                        return resCallback(chunk);
+                    };
+                    return wrappedResFunction.apply(
+                        this,
+                        [resEvent, resPatchedCallback.bind(this)]
+                    );
+                };
+            }
+
+            /**
+             * Wraps 'on' method in a request to capture response event.
+             * @param {Function} wrappedReqFunction The wrapped end function
+             * @returns {Function} The wrapped function
+             */
+            function requestOnWrapper(wrappedReqFunction) { // eslint-disable-line no-inner-declarations,max-len
+                // epsagonMarker is sent only on our call in this module
+                return function internalRequestOnWrapper(reqEvent, reqCallback, epsagonMarker) {
+                    if (
+                        reqEvent !== 'response' ||
+                        epsagonMarker ||
+                        typeof reqCallback !== 'function'
+                    ) {
+                        return wrappedReqFunction.apply(this, [reqEvent, reqCallback]);
+                    }
+                    const reqPatchedCallback = (res) => {
+                        if (res.EPSAGON_PATCH) {
+                            return reqCallback(res);
+                        }
+                        res.EPSAGON_PATCH = true;
```

Comment:
i'd check res is not undefined just to be on the safe side

Reply:
👍